### ISSUE TITLE
RandomFishEye not transforming bboxes

### ISSUE THREAD

**Issue by tynic** [kornia/kornia#1697]
### Describe the bug

When using `RandomFishEye` augmentation on an image and bbox, the output image is transformed however bbox is not.

### Reproduction steps

```bash
import cv2
import numpy as np
from kornia import image_to_tensor
from kornia.augmentation import AugmentationSequential, RandomFisheye
from kornia.constants import DataKey
import torch
from matplotlib import pyplot as plt
from torchvision.transforms import transforms


to_pil = transforms.ToPILImage()

def plot_resulting_image(img, bbox):
    img_draw = cv2.polylines(
        np.array(to_pil(img)), np.int32(bbox.numpy()), isClosed=True, color=(255, 0, 0)
    )
    return img_draw


plt.axis("off")

bbox = torch.tensor([[[355, 10], [660, 10], [660, 250], [355, 250]]])

# read the image
img = cv2.imread("/path/to/panda.jpg", cv2.IMREAD_COLOR)

# plot the image to see whether the bbox is on its place
img_plotted = plot_resulting_image(img, bbox)
plt.imshow(img_plotted)
plt.savefig("/path/to/output/myfig1.png")

img_tensor = image_to_tensor(img).float() / 255.0

# settings for the RandomFishEye
center_x = torch.tensor([-0.3, 0.3])
center_y = torch.tensor([-0.3, 0.3])
gamma = torch.tensor([0.9, 1.0])

aug_list = AugmentationSequential(
    RandomFisheye(center_x, center_y, gamma, p=1.0),
    data_keys=[DataKey.INPUT, DataKey.BBOX],
)

# run augmentation on image and bbox
img_out, boxes_out = aug_list(img_tensor, bbox)

# plot the results
img_out = plot_resulting_image(img_out.squeeze(), boxes_out)
plt.imshow(img_out)
plt.savefig("/path/to/output/myfig2.png")
```

### Expected behavior

The output bbox should be transformed along with the image, instead, the output bbox is never modified:
`myfig1.png`:
![myfig1](https://user-images.githubusercontent.com/7979110/166646084-01578c63-0965-45b5-95a9-8f47f23d5cb3.png)

`myfig2.png`
![myfig2](https://user-images.githubusercontent.com/7979110/166646633-2704e3e7-df98-4a85-b8cb-48aa3a3ef012.png)

### Environment

```
PyTorch version: 1.10.0+cu113
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~18.04) 9.4.0
Clang version: Could not collect
CMake version: version 3.10.2
Libc version: glibc-2.27

Python version: 3.8.10 | packaged by conda-forge | (default, May 11 2021, 07:01:05)  [GCC 9.3.0] (64-bit runtime)
Python platform: Linux-4.15.0-176-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.3.109
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1080 Ti
Nvidia driver version: 465.19.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.7.1
[pip3] mypy==0.770
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.2
[pip3] pytorch-metric-learning==1.0.0
[pip3] torch==1.10.0+cu113
[pip3] torchaudio==0.10.0+cu113
[pip3] torchfile==0.1.0
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.11.1+cu113
[conda] cudatoolkit               11.2.72              h2bc3f7f_0    nvidia
[conda] efficientnet-pytorch      0.7.1                    pypi_0    pypi
[conda] mypy                      0.770                    pypi_0    pypi
[conda] mypy-extensions           0.4.3                    pypi_0    pypi
[conda] numpy                     1.20.1                   pypi_0    pypi
[conda] pytorch-metric-learning   1.0.0                    pypi_0    pypi
[conda] torch                     1.10.0+cu113             pypi_0    pypi
[conda] torchaudio                0.10.0+cu113             pypi_0    pypi
[conda] torchfile                 0.1.0                    pypi_0    pypi
[conda] torchsummary              1.5.1                    pypi_0    pypi
[conda] torchvision               0.11.1+cu113             pypi_0    pypi
```

**Comment by edgarriba** [comment]
@tynic the fisheye transform is a non-rigid geometric transformation, that has a vector field per pixel and limits to transforms coordinates as opposed to the affine group transforms. I currently don't have a direct solution to the problem but worth to explore if you have any idea. /cc @shijianjian @ducha-aiki ?

**Comment by tynic** [comment]
@edgarriba Thank you for your response. In such cases, shouldn't an exception be thrown and the restriction properly documented? Silently returning the original boxes is confusing.

**Comment by shijianjian** [comment]
I think a stupid way of doing it can be:
1. input the bbox as a mask,
2. fit the output with either outer rectangle or inner rectangle, that can be a parameter.

I am wondering how easy it is to have an equation to compute the every output points with the `center_x` and `center_y`. Given
https://github.com/kornia/kornia/blob/1172b8bd3ac239741070db4af572fca0e97ea474/kornia/augmentation/_2d/geometric/fisheye.py#L82-L93

Just by reading the code, given `(x, y)` point, can't we have something similar like:
```
distance = ((center_x - x) ** 2 + (center_y - y) ** 2) ** 0.5
out_x = x + x * distance ** gamma
out_y = y + y * distance ** gamma
```

Probably we might start supporting non-affine transformations.

**Comment by shijianjian** [comment]
This should work to remap the points.

@tynic Can you file a PR to throw a user warning for some augmentations like FishEye and ThinPlate? I think we might need another paradigm to handle such non-affine transformations whilst `AugmentationSequential`.

```python
def fisheye_point(point, x, center_x, center_y, gamma):
    B, _, H, W = x.shape

    field_x = point[0] * 2 / H - 1.
    field_y = point[1] * 2 / W - 1.

    # vectorize the random parameters
    center_x = center_x.view(B, 1, 1).to(x)
    center_y = center_y.view(B, 1, 1).to(x)
    gamma = gamma.view(B, 1, 1).to(x)

    # compute and apply the distances respect to the camera optical center
    distance = ((center_x - field_x) ** 2 + (center_y - field_y) ** 2) ** 0.5
    # field_x = field_x + field_x * distance ** gamma  # BxHxw
    # field_y = field_y + field_y * distance ** gamma  # BxHxW
    # reverse the calculation of specific points.
    field_x = field_x / (1 + distance ** gamma)  # BxHxw
    field_y = field_y / (1 + distance ** gamma)  # BxHxw
    return field_x, field_y
```

![image](https://user-images.githubusercontent.com/15955486/166976309-3d9efc64-4c23-4ffa-81ce-7d9c1aec0933.png)

**Comment by shijianjian** [comment]
@tynic can you post the full script to reproduce your results?

**Comment by wadoodislam** [comment]
@tynic I have used the above implementation but I think you are using height to normalize the x coordinates and width for normalizing y coordinates.

**Comment by edgarriba** [comment]
If anyone is interested in this, feel free to re-open the ticket. Thanks!
